diff --git a/lib/pico-sdk b/lib/pico-sdk
--- a/lib/pico-sdk
+++ b/lib/pico-sdk
@@ -1 +1 @@
-Subproject commit 2e6142b15b8a75c1227dd3edbe839193b2bf9041
+Subproject commit 2e6142b15b8a75c1227dd3edbe839193b2bf9041-dirty
diff --git a/ports/rp2/rp2_pio.c b/ports/rp2/rp2_pio.c
index eb85eb224..2f107d251 100644
--- a/ports/rp2/rp2_pio.c
+++ b/ports/rp2/rp2_pio.c
@@ -114,6 +114,11 @@ void rp2_pio_init(void) {
     // Set up interrupts.
     memset(MP_STATE_PORT(rp2_pio_irq_obj), 0, sizeof(MP_STATE_PORT(rp2_pio_irq_obj)));
     memset(MP_STATE_PORT(rp2_state_machine_irq_obj), 0, sizeof(MP_STATE_PORT(rp2_state_machine_irq_obj)));
+
+    irq_set_mask_enabled((1u << PIO0_IRQ_0) | (1u << PIO0_IRQ_1), false);
+    irq_remove_handler(PIO0_IRQ_0, irq_get_exclusive_handler(PIO0_IRQ_0));
+    irq_remove_handler(PIO0_IRQ_1, irq_get_exclusive_handler(PIO0_IRQ_1));
+
     irq_set_exclusive_handler(PIO0_IRQ_0, pio0_irq0);
     irq_set_exclusive_handler(PIO1_IRQ_0, pio1_irq0);
 }
